src/pharmpy/tools/external/nonmem/run.py

```diff
@@ -22,6 +22,11 @@
 def execute_model(model_or_model_entry, db):
     if isinstance(model_or_model_entry, ModelEntry):
         model = model_or_model_entry.model
+
+        # TODO : Remove after complete integration of ModelEntry
+        if model.modelfit_results is None:
+            model = model.replace(modelfit_results=None)
```

Comment:
If `modelfit_results` was checked to be `None` why set it to `None` again?

Reply:
Oh no this is incorrect. The if statement was added to avoid performing the operations when they aren't needed but it should be `if model.modelfit_results is not None:`
Great catch, I'll fix it right away